File: CustomBroadcastReceiverApp/app/src/main/java/com/example/custombroadcastreceiverapp/MainActivity.java
package com.example.custombroadcastreceiverapp;

import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;


public class MainActivity extends AppCompatActivity {
    private MyCustomBroadcastReceiver myReceiver;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        myReceiver = new MyCustomBroadcastReceiver();
        Button sendBroadcastButton =
                findViewById(R.id.sendBroadcastButton);
        sendBroadcastButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Create an intent with the custom action
                Intent intent = new Intent("com.example.CUSTOM_ACTION");
                // Send the broadcast
                LocalBroadcastManager.getInstance(MainActivity.this).sendBroadcast(intent);
            }
        });
    }
    @Override
    protected void onStart() {
        super.onStart();
        // Register the receiver with the intent filter
        LocalBroadcastManager.getInstance(this)
                .registerReceiver(myReceiver, new
                        IntentFilter("com.example.CUSTOM_ACTION"));
    }
    @Override
    protected void onStop() {
        super.onStop();
        // Unregister the receiver to prevent memory leaks

        LocalBroadcastManager.getInstance(this).unregisterReceiver(myReceiver);
    }
}

File: CustomBroadcastReceiverApp2/app/src/main/java/com/example/custombroadcastreceiverapp2/MainActivity.kt
import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import kotlinx.android.synthetic.main.activity_main.*
class MainActivity : AppCompatActivity() {
    private lateinit var myReceiver: MyCustomBroadcastReceiver
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        myReceiver = MyCustomBroadcastReceiver()
        sendBroadButton.setOnClickListener {
            // Create an intent with the custom action
            val intent = Intent("com.example.CUSTOM_ACTION")
            // Send the broadcast
            LocalBroadcastManager.getInstance(this).sendBroadcast(intent)
        }
    }
    override fun onStart() {
        super.onStart()
        // Register the receiver with the intent filter
        LocalBroadcastManager.getInstance(this)
            .registerReceiver(myReceiver,
                IntentFilter("com.example.CUSTOM_ACTION"))
    }
    override fun onStop() {
        super.onStop()
        // Unregister the receiver to prevent memory leaks

        LocalBroadcastManager.getInstance(this).unregisterReceiver(myReceiver)
    }
}
